File: code/helpers/helpers_test.py
import unittest

from .helpers import read_file, read_file_line_by_line

class TestHelpers(unittest.TestCase):
    def test_read_file(self) -> None:
        self.assertEqual(
            read_file(filepath="code/helpers/helpers.txt"),
            "this\nis\na\ntest\n"
        )
    def test_read_file_line_by_line(self) -> None:
        self.assertEqual(
            read_file_line_by_line(filepath="code/helpers/helpers.txt"),
            ["this", "is", "a", "test"]
        )

File: code/helpers/helpers_test.go
package main

import "testing"

func TestReadFile(t *testing.T) {
	input := "helpers.txt"
	expectedOutput := "this\nis\na\ntest\n"
	calculatedOutput, err := ReadFile(input)
	if err != nil {
		t.Errorf("%s", err)
	}
	if expectedOutput != calculatedOutput {
		t.Errorf("Output mismatch. Expected %s. Got %s", expectedOutput, calculatedOutput)
	}
}

func TestReadFileLineByLine(t *testing.T) {
	input := "helpers.txt"
	expectedOutput := []string{"this", "is", "a", "test"}
	calculatedOutput, err := ReadFileLineByLine(input)
	if err != nil {
		t.Errorf("%s", err)
	}
	for index := range expectedOutput {
		if expectedOutput[index] != calculatedOutput[index] {
			t.Errorf("Output mismatch. Expected %s. Got %s", expectedOutput[index], calculatedOutput[index])
		}
	}

}
